hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/io/HoodieSparkFileGroupReaderBasedMergeHandle.java

```diff
@@ -191,7 +191,7 @@ public void write() {
         writeSchemaWithMetaFields,
         internalSchemaOption,
         hoodieTable.getMetaClient(),
-        hoodieTable.getMetaClient().getTableConfig().getProps(),
+        config.getProps(),
```

Comment:
We may want to merge the table config and write config and then pass in the merged props, because the write configs corresponding to the table configs may not be required and the file group reader prefers table config.